pkg/pkgmgr/dpkg.go

```diff
@@ -499,9 +521,31 @@ func (dm *dpkgManager) unpackAndMergeUpdates(ctx context.Context, updates unvers
 	copyStatusCmd := fmt.Sprintf(strings.ReplaceAll(copyStatusTemplate, "\n", ""), dpkgStatusFolder, dm.statusdNames)
 	statusUpdated := fieldsWritten.Dir(resultsPath).Run(llb.Shlex(copyStatusCmd)).Root()
 
+	// If the image has been patched before, diff the base image and patched image to retain previous patches
+	if dm.config.PatchedConfigData != nil {
+		// Diff the base image and patched image to get previous patches
+		prevPatchDiff := llb.Diff(dm.config.ImageState, dm.config.PatchedImageState)
+
+		// Diff the manifests for the latest updates
+		manifestsDiff := llb.Diff(fieldsWritten, statusUpdated)
```

Comment:
how come manifest diff needs to be done sparately? why isnt the manifest diff reflected within the normal patch diff files?